src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java

```diff
@@ -427,17 +429,19 @@ public int getCount() throws DaoException {
      * Deletes all Genetic Alteration Records associated with the specified Genetic Profile ID.
      *
      * @param geneticProfileId Genetic Profile ID.
+     * @param geneticEntityId Genetic Entity ID.
      * @throws DaoException Database Error.
      */
-    public void deleteAllRecordsInGeneticProfile(long geneticProfileId) throws DaoException {
+    public void deleteAllRecordsInGeneticProfile(long geneticProfileId, long geneticEntityId) throws DaoException {
```

Comment:
maybe `deleteAllGeneticAlterationRecordsInGeneticProfile` which then deletes all data (not just for one entity) instead of deleting and inserting one by one...

Reply:
I did it like you say at the end https://github.com/cBioPortal/cbioportal-core/blob/rfc79-feedback/src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java#L47

Reply:
@pieterlukasse My bad, I've mixed `deleteAllRecordsInGeneticProfile` and `deleteAllSamplesInGeneticProfile`. We are still delete those one-by-one here https://github.com/cBioPortal/cbioportal-core/blob/rfc79-feedback/src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java#L52 and here https://github.com/cBioPortal/cbioportal-core/blob/rfc79-feedback/src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java#L59.

Let me address this.

Reply:
here is the change https://github.com/cBioPortal/cbioportal-core/pull/47/commits/ab51a4b601dfdb6df753e8a6d58e048b45967269